index.md

```diff
@@ -8,6 +8,573 @@ This is some welcome text.
+attempt to invoke what it believes to be a method, which will fail since
+it has become a field instead (a variable). The syntax will error out, and
+this can be particularly dangerous when masking the function when a parent
+class relies on calling the method.
+
+```python
+class Example(object):
+    def field(self, num):
+        return num
+    def __init__(self):
+        self.field = 'Masking the function with this string'
+
+# If you call Example().field(num), it will error out since we masked it
+```
+
+
+
+### E0203: Access to member before definition {#E0203}
+
+Before trying to use a member of a class, it should have been defined at
+some point. If you try to use it before assigning to it, Python cannot
+resolve the value and an error will occur.
+
+```python
+class MyClass:
+	def __init__(self):
+		print(self.a)  # Haven't defined self.a yet, can't use
+		self.a = 5
+```
+
+
+
+### E0211: No method argument {#E0211}
+
+Each method in a class needs to have at least one argument (which is usually
+_self_). Python uses this to call methods, and the first argument is populated
+with the object that is calling the method. This is what allows you to access
+the calling object.
+
+For example, the following two are equivalent:
+
+```python
+class A:
+	def __init__(self):
+		pass
+	def method(self):
+		print('Hi')
+
+# First:
+a = A()
+a.method()
+
+# Second:
+a = A()
+A.method(a)
+```
+
+Therefore, if you do not provide any arguments, then Python does not know
+how to pass the object to the method, and it will error out. To fix this,
+put _self_ (or some variable name) in the parenthesis for the method call.
+
+```python
+class MyClass:
+	def __init__(self):
+		pass
+	def method():
+		print('Missing argument for method definition')
+```
+
+
+
+### E0213: Self as the first argument {#E0213}
+
+The first argument should be the exact word 'self'. This is not an error,
+but it's such a common practice that this is considered an error. The
+following is an example of a good, and bad example:
+
+```python
+class MyClass:
+	def __init__(self):
+		pass
+	
+	def methodA():  # Bad
+		pass
+
+	def methodB(self):  # Good
+		pass
+```
+
+
+
+### E0239: Inheriting from a non-class {#E0239}
+
+When you inherit, it must come from a class. If you use something that is
+not a class, you won't be able to inherit from it. If you want to inherit
+from the type of object you are providing, look up its class with the
+__class__ field.
+
+```python
+class newclass("str"):
+	pass
+```
+
+
+
+### E0241: Duplicate bases {#E0241}
+
+When inheriting, you should only specify a class once to inherit from,
+multiple times is an error:
+
+```python
+class A:
+	pass
+	
+class B(A, A):  # Only include A once to inherit properly
+	pass
+```
+
+
+
+### E0302: Unexpected special method signature {#E0302}
+
+Occurs when a special method (has underscores on both sides) does not have
+the expected number of arguments, which likely will lead to errors somewhere.
```

Comment:
That last phrase is too vague. Perhaps mention that these are overriding existing built-in methods, which Python expects to use with a certain signature. Overriding methods shouldn't change signatures.